arch/avr/src/avr/avr_switchcontext.c

```diff
@@ -85,6 +85,10 @@ void up_switch_context(struct tcb_s *tcb, struct tcb_s *rtcb)
 
       nxsched_resume_scheduler(tcb);
 
+      /* Record the new "running" task */
+
+      g_running_tasks[this_cpu()] = tcb;
+
```

Comment:
@hujun260 I think we need to test this modification on real hardware (i.e. for 8-bit MCU like AVR8 here)

Reply:
Unfortunately, I don't have this hardware on my side.

But based on the code analysis, it seems that avr_switchcontext does not update g_running_tasks, which definitely means that this logic has been overlooked.